### ISSUE TITLE
GOG SkyrimSE Profile issue

### ISSUE THREAD

**Issue by dizietsma** [Nexus-Mods/Vortex#13413]
I've noticed the following in my vortex.log (edited for username):

Fri, 28 Oct 2022 17:17:24 GMT - debug: sync from profile profilePath=C:\Users\MYUSER\AppData\Roaming\Vortex\skyrimse\profiles\rWVGTvXbc, sourceFiles=[C:\Users\MYUSER\AppData\Local\Skyrim Special Edition\plugins.txt, C:\Users\MYUSER\AppData\Local\Skyrim Special Edition\loadorder.txt]
Fri, 28 Oct 2022 17:17:24 GMT - debug: sync from profile complete

The problem is that the GOG version of SkyrimSE seems to be using:
C:\Users\MYUSER\AppData\Local\Skyrim Special Edition GOG\plugins.txt

this would appear to answer my two previous attempts to report the issue with profiles via the feedback mechanism, namely that plugins.txt and loadorder.txt are not saved in the profiles folders for this game. (Or perhaps they are, but since the file never changes due to the game no longer using them in that location, the effect is the same.)

Even removing the old AppData\Local\Skyrim Special Edition folder does not cause any visible error in Vortex.  This profile issue has indeed been happening since I moved to the GOG version of SkyrimSE and updated to 1.6.14.
What does happen is that the profile folders then get 0 KB versions of these files.

diziet

**Comment by MrGrymReaper** [comment]
I have something similar to yours in that it is not utilising the correct paths in the user's AppData folder, in the case of GOG Skyrim SE. It doesn't even utilise or provide a set of files in the GOG Skyrim folder, which is where the game utilises it from, thus the game likely either generates again itself at some point (with incorrect or out of updated sync). Alternatively, it just bombs out suddenly and crashes to the desktop as it can't find it.

To fix this issue (bug) as well as anticipate other tools which utilise this path (e.g. xEdit), how about having it create some kind of configurable link (e.g. possibly hard link or symlink etc) which points to these. That go somelink link the following:-

Link name (1): C:\Users\<username>\AppData\Local\Skyrim Special Edition\plugins.txt
Link target (1): C:\Users\<username>\AppData\Local\Skyrim Special Edition GOG\plugins.txt
Link name (2): C:\Users\<username>\AppData\Local\Skyrim Special Edition\loadorder.txt
Link target (2): C:\Users\<username>\AppData\Local\Skyrim Special Edition GOG\loadorder.txt

Another possible fix solution to this issue (bug) is to have the Vortex mod manager directly identify if an SkyrimSE.exe is Steam (original PC store) based, or alternatively the new GOG Skyrim SE. As currently the Vortex mod manager is unable to detect the GOG releases. It currently occurs when installing with offline installers and with the GOG Galaxy client.

Then automatically switch to utilising the C:\Users\MYUSER\AppData\Local\Skyrim Special Edition GOG\plugins.txt and C:\Users\MYUSER\AppData\Local\Skyrim Special Edition GOG\loadorder.txt files. My preference is for utilising hard links when possible, during the establishing of management of the Skyrim SE on Vortex.

**Comment by dizietsma** [comment]
This issue still exists as of version 1.7.2.  It would be useful to know if any user of the GOG version of SkyrimSE does NOT have this issue; and whether they installed on a system with no Skyrim or installed the GOG version and then changed the location in Vortex from the Steam version to GOG, as I did.

diziet

**Comment by TanninOne** [comment]
@dizietsma I've not been able to reproduce this at all. I've had several versions of SSE installed (Steam, GOG, Game Pass), all at once for a while and if Vortex is set to manage a version it updates the correct plugin file.

This does however rely on Vortex actually knowing which store the game was installed through.
That works if you're using the version that was autodetected without manually setting the location or, if you do manually set the location, Vortex will compare the selected path against the path identified by the various stores and if that doesn't work either, the gog version would be identified by the existence of a gog.ico file.
I can see the path comparison fail, e.g. if you use directory junction points or something to have the files appear on one drive while actually being stored somewhere else but for the icon file check to fail you'd have to intentionally delete it afaik.

If you go to the "Games" tab and click the "..." button on SSE, what _does_ it say under "Game Store"?

**Comment by dizietsma** [comment]
Hi, thanks for getting back:)
All game files and staging and download files for SSE are on the same 'D:' drive, with appdata being on the 'C:' drive, there are no junction points for SSE (Witcher 3 is a different story!)
It does say GOG as the game store, and the is a gog.ico file.  When I installed the GOG version I already had the steam version with Vortex managing it.  I uninstalled the steam version and manually specified the new location. I can't remember if this was before or after Vortex began recognising GOG SkyrimSE.  I noticed at the time that profiles didn't come across completely, savegames didn't show up although the profile folders were still there, perhaps I had to make new profiles and copy saves across, again I can't recall. :(
I'm wondering if this isn't a symptom of migration rather than how Vortex supports a fresh GOG install.  I have been toying with the idea of doing a once only complete refresh from a system with no SSE at all and no management or profiles for SSE in Vortex.  Knowing that it _can_ be done is good, though it would be nice to avoid since I have mods that are not on Nexus or I made myself that I cannot just add to a collection for simple reinstall!
Is there a way to force the detection and have Vortex setup the profile ini files again?
Could I remove management of SSE from Vortex but keep the profiles and then remanage?  Perhaps using a backup?

diziet

**Comment by TanninOne** [comment]
Please don't do anything drastic to your setup, there shouldn't be anything "magical" or hidden in what Vortex does that could be affected by migration (not in 1.7 at least).

This was a bit more complex in 1.6.x but in 1.7 this is much simplified, there is just a flag for each game discovered telling us which store we associate it with. When the ini files are written, we look at that flag and nothing else to determine where the plugins.txt/loadorder.txt need to go, those paths aren't stored but determined on the fly so it can't be outdated.

If the games tab correctly identifies the store as the "gog" variant this should just work (tm). I can't think of anything you could have done wrong or that could have gone wrong in the past for it to fail now, there has to be a bug. Maybe if you reset enough you could get to a point where the bug won't trigger for you anymore but it'd still be there.

I will add a bit of logging to the next release (1.7.3) to narrow down where it goes wrong, my preference would be you live with the problem for now.

**Comment by TanninOne** [comment]
@dizietsma 1.7.3 is out, please run again (this will likely not fix anything) and then send me a current log file

**Comment by dizietsma** [comment]
Well oddly it does seem to have fixed things, at least I've been swapping between profiles - one with a few mods and the other with lots - and the mods and plugins seem to be properly en/disabled.
Logs attached:


diziet

**Comment by TanninOne** [comment]
Yeah, seems to be fine based on the log. Except that one of those log messages is generated waaaaay too often.. ;)

**Comment by TanninOne** [comment]
fixed, as far as I know
